compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/GeneratorUtils.java

```diff
@@ -93,6 +98,52 @@ public static String getDeprecationReason(Documentable documentable) {
         return documentable.documentation().get().deprecatedDescription().get();
     }
 
+    public static String removeEscapeCharacter(String identifier) {
+        if (identifier == null) {
+            return null;
+        }
+
+        Matcher matcher = UNICODE_PATTERN.matcher(identifier);
+        StringBuffer buffer = new StringBuffer(identifier.length());
+        while (matcher.find()) {
+            String leadingSlashes = matcher.group(1);
+            if (isEscapedNumericEscape(leadingSlashes)) {
+                // e.g. \\u{61}, \\\\u{61}
+                continue;
+            }
+
+            int codePoint = Integer.parseInt(matcher.group(2), 16);
+            char[] chars = Character.toChars(codePoint);
+            String ch = String.valueOf(chars);
+
+            if (ch.equals("\\")) {
+                // Ballerina string unescaping is done in two stages.
+                // 1. unicode code point unescaping (doing separately as [2] does not support code points > 0xFFFF)
+                // 2. java unescaping
+                // Replacing unicode code point of backslash at [1] would compromise [2]. Therefore, special case it.
+                matcher.appendReplacement(buffer, Matcher.quoteReplacement(leadingSlashes + "\\u005C"));
+            } else {
+                matcher.appendReplacement(buffer, Matcher.quoteReplacement(leadingSlashes + ch));
+            }
+        }
+        matcher.appendTail(buffer);
+        String value = String.valueOf(buffer);
+
+        if (value.startsWith("'")) {
```

Comment:
I think we can extract `"'"` to a common constant and mark it as `BALLERINA_ESCAPE_CHARACTER` or something which gives that meaning.

Reply:
Fixed with 9df6124e904a963cb8f8995402f55c15f18836f7